### PR TITLE
@86epxqyub #Create regular question step 2 and validation

### PR DESCRIPTION
**Pull request by DropitupQwerty**
- Create regular question questionnaire validation
- Fix default values bug on answer options

### CODE REVIEW

File: packages/core-library/components/Radio/Radio.tsx [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    RadioGroup as MuiRadioGroup,
+    RadioGroupProps,
+    RadioProps as MuiRadioProps,
+    TypographyProps,
+    Typography,
+} from '@mui/material';
+import { Controller, FieldValues } from 'react-hook-form';
+import { ControlledField } from '../Textfield';
+
+
```
Comment (DropitupQwerty) [inline packages/core-library/components/Radio/Radio.tsx:14]: Spacing

File: packages/core-library/components/Radio/Radio.tsx [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    labelProps?: TypographyProps;
+} & Pick<MuiRadioProps, 'color'>;
+
+const sx: MuiRadioProps['sx'] = {
+    color: 'black',
+    '&.Mui-checked': {
+        color: 'black',
+    },
+};
+
```
Comment (DropitupQwerty) [inline packages/core-library/components/Radio/Radio.tsx:29]: Spacijg

File: packages/core-library/components/blocks/AnswerOptions/blocks/SATA/SATA.tsx [hunk 3]
```diff
@@ -1,9 +1,9 @@
...
 import DeleteOutlineIcon from "@mui/icons-material/DeleteOutline";
 
-import { useFieldArray } from "react-hook-form";
+import { useFieldArray, useFormContext } from "react-hook-form";
 import AddIcon from "@mui/icons-material/Add";
-import { IconButton } from "@mui/material";
+import { Box, IconButton, Typography } from "@mui/material";
 import { ContainedRegularQuestionType } from "../../../../../system/app/internal/blocks/Hub/Settings/SettingsManagement/steps/content/simulator/types";
 import { Button, Card, ControlledCheckbox } from "../../../..";
 import { ControlledTextField } from "../../../../Textfield/TextField";
```
Comment (DropitupQwerty) [inline packages/core-library/components/blocks/AnswerOptions/blocks/SATA/SATA.tsx:9]: Should use absolute importing

File: packages/core-library/system/app/internal/blocks/Hub/Settings/SettingsManagement/constants/constants.ts [hunk 4]
```diff
@@ -1,7 +1,21 @@
...
-  cognitiveLevel: "",
-  contentArea: "",
-  answers: Array(2).fill({ answer: "", answerKey: false }),
+import { RegularQuestionSelectionOptions } from '../types';
+
+export const initAnswerValues = { answer: "", answerKey: false };
+
+export const initQuestionsValues = (questionnaireType: RegularQuestionSelectionOptions | undefined) => {
+
+
```
Comment (DropitupQwerty) [inline packages/core-library/system/app/internal/blocks/Hub/Settings/SettingsManagement/constants/constants.ts:7]: Spacing

**Review by JezChristian — APPROVED** [review]

**Comment by JezChristian** [comment]
Code Cov Build Failure

**Review by acostales08 — APPROVED** [review]

File: packages/core-library/components/blocks/AnswerOptions/blocks/MCQ/MCQ.tsx [hunk 5]
```diff
@@ -1,12 +1,76 @@
-import React from 'react'
+import React, { useEffect, useState } from "react";
+import { useFormContext } from "react-hook-form";
+import { Box, RadioGroup } from "@mui/material";
+import { ContainedRegularQuestionType } from "../../../../../system/app/internal/blocks/Hub/Settings/SettingsManagement/steps/content/simulator/types";
+import { ControlledTextField } from "../../../../Textfield/TextField";
+import { ControlledRadio, Card } from '../../../../';
```
Comment (JezChristian) [inline packages/core-library/components/blocks/AnswerOptions/blocks/MCQ/MCQ.tsx:6]: To lessen the lines, maybe you can merge the import of ControlledTextField , ControlledRadio and Card

Comment (JezChristian) [inline packages/core-library/components/blocks/AnswerOptions/blocks/MCQ/MCQ.tsx:6]: And the path somehow feels wrong to be "../../../../"

File: packages/core-library/tests/components/Blocks/AnswerOptions/AnswerOptions.test.tsx [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+describe('AnswerOptions', () => {
+    const { result } = renderHook(() => useForm())
+    const form = result.current
+
+    it('should render the mcq answer options', () => {
+        render(<FormProvider {...form}><AnswerOptions {...DEFAULT_PROPS} questionnaireType='MCQ' /></FormProvider>)
+        expect(screen.getByTestId('mcq-answer')).toBeInTheDocument()
+    })
+
+
```
Comment (JezChristian) [inline packages/core-library/tests/components/Blocks/AnswerOptions/AnswerOptions.test.tsx:31]: spacing

File: packages/core-library/tests/components/Blocks/AnswerOptions/AnswerOptions.test.tsx [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+        render(<FormProvider {...form}><AnswerOptions {...DEFAULT_PROPS} questionnaireType='SATA' /></FormProvider>)
+        expect(screen.getByTestId('sata-answer')).toBeInTheDocument()
+    })
+
+    it('should render nothing ', () => {
+        const { container } = render(<FormProvider {...form}><AnswerOptions questionIndex={1} questionType='caseStudy' questionnaireType='SATA' /></FormProvider>)
+        expect(container).toBeEmptyDOMElement()
+    })
+
+
```
Comment (JezChristian) [inline packages/core-library/tests/components/Blocks/AnswerOptions/AnswerOptions.test.tsx:42]: spacing

File: packages/core-library/tests/components/Blocks/AnswerOptions/AnswerOptions.test.tsx [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+    it('should render nothing when questionType is caseStudy and MCQ', () => {
+        const { container } = render(<FormProvider {...form}><AnswerOptions questionIndex={1} questionType='caseStudy' questionnaireType='MCQ' /></FormProvider>)
+        expect(container).toBeEmptyDOMElement()
+    })
+
```
Comment (JezChristian) [inline packages/core-library/tests/components/Blocks/AnswerOptions/AnswerOptions.test.tsx:47]: spacing

File: packages/core-library/tests/components/Blocks/AnswerOptions/blocks/SATA.test.tsx [hunk 9]
```diff
@@ -0,0 +1,132 @@
+import { FormProvider, useFieldArray, useForm, useFormContext } from 'react-hook-form';
+import { SATA } from '../../../../../components/blocks/AnswerOptions/blocks/SATA/SATA';
+import { fireEvent, renderHook, screen } from '../../../../common'
+import { getAllByRole, render, waitFor } from '@testing-library/react';
+import { ContainedRegularQuestionType } from '../../../../../system/app/internal/blocks/Hub/Settings/SettingsManagement/steps/content/simulator/types';
+import { initAnswerValues, initQuestionsValues } from '../../../../../system/app/internal/blocks/Hub/Settings/SettingsManagement/constants/constants';
+
+
+
```
Comment (JezChristian) [inline packages/core-library/tests/components/Blocks/AnswerOptions/blocks/SATA.test.tsx:9]: spacing

File: packages/core-library/tests/components/Blocks/AnswerOptions/blocks/SATA.test.tsx [hunk 10]
```diff
@@ -0,0 +1,132 @@
...
+jest.mock("react-hook-form", () => ({
+    ...jest.requireActual("react-hook-form"),
+    useFieldArray: jest.fn(),
+    useFormContext: jest.fn(),
+}));
+
+
+const MOCK_QUESTIONTYPE = "SATA"
+
+
```
Comment (JezChristian) [inline packages/core-library/tests/components/Blocks/AnswerOptions/blocks/SATA.test.tsx:30]: spacing

File: packages/core-library/tests/components/Blocks/AnswerOptions/blocks/SATA.test.tsx [hunk 11]
```diff
@@ -0,0 +1,132 @@
...
+    useRouter: jest.fn(),
+}));
+
+jest.mock("react-hook-form", () => ({
+    ...jest.requireActual("react-hook-form"),
+    useFieldArray: jest.fn(),
+    useFormContext: jest.fn(),
+}));
+
+
```
Comment (JezChristian) [inline packages/core-library/tests/components/Blocks/AnswerOptions/blocks/SATA.test.tsx:27]: spacing

File: packages/core-library/tests/components/Blocks/AnswerOptions/blocks/SATA.test.tsx [hunk 12]
```diff
@@ -0,0 +1,132 @@
...
+            append: mockAppend,
+            remove: mockRemove,
+        });
+    });
+
+    afterEach(() => {
+        jest.resetAllMocks();
+    });
+
+
```
Comment (JezChristian) [inline packages/core-library/tests/components/Blocks/AnswerOptions/blocks/SATA.test.tsx:50]: spacing

**Review by JezChristian — COMMENTED** [review]
More on spacing , kindly remove the unnecessary spacing, Thank you!

File: packages/core-library/components/blocks/AnswerOptions/blocks/MCQ/MCQ.tsx (hunk 5 above)
Comment (DropitupQwerty) [reply packages/core-library/components/blocks/AnswerOptions/blocks/MCQ/MCQ.tsx:6]: Having a build build failure when yung using core-library import

Comment (JMSevilla) [reply packages/core-library/components/blocks/AnswerOptions/blocks/MCQ/MCQ.tsx:6]: We can accept this for now

File: packages/core-library/components/blocks/AnswerOptions/blocks/MCQ/MCQ.tsx [hunk 13]
```diff
@@ -1,12 +1,76 @@
...
+        <Box
+          display="flex"
+          flexDirection="column"
+          minHeight="200px"
+          maxHeight="400px"
+          overflow="auto"
+          padding={3}
+          gap={1}
+        >
+          {Array.from({ length: 4 }).map((_, index) => (
```
Comment (JMSevilla) [inline packages/core-library/components/blocks/AnswerOptions/blocks/MCQ/MCQ.tsx:54]: Suggestion for this

declare constant
const NUMBER_OF_OPTIONS = 4;

then > Array.from({ length: NUMBER_OF_OPTIONS }).map((_, index) =>

File: packages/core-library/tests/components/Blocks/AnswerOptions/blocks/SATA.test.tsx [hunk 14]
```diff
@@ -0,0 +1,120 @@
+import { FormProvider, useFieldArray, useForm, useFormContext } from 'react-hook-form';
+import { SATA } from '../../../../../components/blocks/AnswerOptions/blocks/SATA/SATA';
+import { fireEvent, renderHook, screen } from '../../../../common'
+import { getAllByRole, render, waitFor } from '@testing-library/react';
```
Comment (JMSevilla) [inline packages/core-library/tests/components/Blocks/AnswerOptions/blocks/SATA.test.tsx:4]: We should still use this imports from our commons

**Review by JMSevilla — CHANGES_REQUESTED** [review]

**Review by BryanPalad — APPROVED** [review]

**Review by JMSevilla — APPROVED** [review]

**Pull request merged**